### PR TITLE
Repair instance of leaving in battle

### PR DESCRIPTION
**Pull request by pklloveyou**
Repair instance of leaving in battle #62

### CODE REVIEW

**Comment by pklloveyou** [comment]
No one seems to need it

File: src/AutoBalance.cpp [hunk 1]
```diff
@@ -407,8 +407,38 @@ class AutoBalance_AllMapScript : public AllMapScript
...
+            }
+            //pklloveyou天鹿:
+            if (AutoBalanceCheck)
+            {
+                for (Map::PlayerList::const_iterator itr = pl.begin(); itr != pl.end(); ++itr)
+                {
+                    if (Player* plr = itr->GetSource())
+                    {
+                        plr->GetSession()->SendNotification("非战斗中进出|cff4cff00%s|r，可以解除锁定", map->GetMapName());
+                        plr->GetSession()->SendNotification("玩家 |cffffffff[%s]|r 在战斗中离开了|cff4cff00%s|r，副本弹性锁定", player->GetName().c_str(), map->GetMapName());
```
Comment (Helias) [inline src/AutoBalance.cpp:433]: What about for enus?

**Comment by Helias** [comment]
Man, sorrry, but maybe who used this didn't receive the notification about this PR.
From my point of view this is useful, we should merge it.

What about the chinese text? could we put enus for an international users..?

File: src/AutoBalance.cpp [hunk 2]
```diff
@@ -407,8 +407,38 @@ class AutoBalance_AllMapScript : public AllMapScript
...
+            //mapABInfo->playerCount--;// (maybe we've to found a safe solution to avoid player recount each time)
             // mapABInfo->playerCount = map->GetPlayersCountExceptGMs();
+            //pklloveyou天鹿:
+            if (map->GetEntry() && map->GetEntry()->IsDungeon())
+            {
+            bool AutoBalanceCheck = false;
+            Map::PlayerList const& pl = map->GetPlayers();
+            for (Map::PlayerList::const_iterator itr = pl.begin(); itr != pl.end(); ++itr)
+            {
+                if (Player* plr = itr->GetSource())
```
Comment (Helias) [inline src/AutoBalance.cpp:419]: indentation

**Review by Helias — CHANGES_REQUESTED** [review]

File: src/AutoBalance.cpp (hunk 1 above)
Comment (pklloveyou) [reply src/AutoBalance.cpp:433]: My English is poor, please help me

Comment (Helias) [reply src/AutoBalance.cpp]: Google translator suggests the following:
`Access can be unlocked during non-combat`
`The player left during the battle. Instance elastic lock` (may we can add `enabled` at the end?)

**Review by Helias — APPROVED** [review]

**Pull request merged**